test/metadata_test.rb

```diff
@@ -70,25 +70,42 @@ class MetadataTest < Minitest::Test
     end
 
     describe "when auth requests are signed" do
-      let(:cert_node) do
-        REXML::XPath.first(
+      let(:key_descriptors) do
+        REXML::XPath.match(
+          xml_doc,
+          "//md:KeyDescriptor",
+          "md" => "urn:oasis:names:tc:SAML:2.0:metadata"
+        )
+      end
+      let(:cert_nodes) do
+        REXML::XPath.match(
           xml_doc,
           "//md:KeyDescriptor/ds:KeyInfo/ds:X509Data/ds:X509Certificate",
           "md" => "urn:oasis:names:tc:SAML:2.0:metadata",
           "ds" => "http://www.w3.org/2000/09/xmldsig#"
         )
       end
-      let(:cert)  { OpenSSL::X509::Certificate.new(Base64.decode64(cert_node.text)) }
+      let(:cert)  { OpenSSL::X509::Certificate.new(Base64.decode64(cert_nodes[0].text)) }
 
       before do
-        settings.security[:authn_requests_signed] = true
         settings.certificate = ruby_saml_cert_text
       end
 
-      it "generates Service Provider Metadata with X509Certificate" do
+      it "generates Service Provider Metadata with AuthnRequestsSigned" do
+        settings.security[:authn_requests_signed] = true
         assert_equal "true", spsso_descriptor.attribute("AuthnRequestsSigned").value
         assert_equal ruby_saml_cert.to_der, cert.to_der
       end
+
+      it "generates Service Provider Metadata with X509Certificate for sign and encrypt" do
+        assert_equal 2, key_descriptors.length
```

Comment:
if there were 3... would that change anything in this spec? otherwise, we don't need the line